workshop/4-Messaging/4.2-Braze.ipynb

```diff
@@ -0,0 +1,522 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### How Does the Template Use the Returned Results?\n",
+    "\n",
+    "Once the `result` variable is populated with the above JSON, the template can reference the product data in the variable using the Braze Liquid templating language.\n",
+    "\n",
+    "A product cell in the template might look like this:\n",
+    "\n",
+    "```\n",
+    "<table width=\"100%\" border=\"0\" cellspacing=\"0\" cellpadding=\"0\">\n",
+    "\t<tr>\n",
+    "\t\t<td width=\"115\" class=\"text3\" style=\"color:#212123; font-family:'Noto Sans', Arial,sans-serif; font-size:18px; line-height:22px; text-align:left\">{{result[1].product.name}}</td>\n",
+    "\t\t<td class=\"img\" style=\"font-size:0pt; line-height:0pt; text-align:left\" width=\"10\"></td>\n",
+    "\t\t<td class=\"text4\" style=\"color:#6e6e70; font-family:'Noto Sans', Arial,sans-serif; font-size:12px; line-height:22px; text-align:right\">{{result[1].product.price}}</td>\n",
+    "\t</tr>\n",
+    "</table>\n",
+    "```\n",
+    "\n",
+    "Note the `result[1].product.name` variable will be replaced with the name of the product that is first in the list of products returned, and the same goes for `result[1].product.price` for that product's price.  The same can be done for other product attributes such as the product's image."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Creating a Braze Campaign\n",
+    "\n",
+    "Log in to the Braze [console](https://dashboard.braze.com/auth).  You will need to have a Braze account configured for you to continue this workshop, or use your existing Braze account.\n",
+    "\n",
+    "![Braze Login Screen](images/braze-create-template-0.png)\n",
+    "\n",
+    "On the left hand pane of the Braze console, click the Templates & Media link.  This will show the Templates list screen.  Click the From File button under Basic Email Templates at the top of the screen.\n",
+    "\n",
+    "![](images/braze-create-template-1.png)\n",
+    "\n",
+    "Click the Upload from File button:\n",
+    "\n",
+    "![](images/braze-create-template-2.png)\n",
+    "\n",
+    "Navigate to your local clone of this repository, to the `/workshop/4-Messaging/braze-templates` directory.  \n",
+    "\n",
+    "Select the `braze-connected-content-email-template.zip` file you created earlier and click the Open or Save button to upload the ZIP file to Braze.\n",
```

Comment:
When running the notebook in SM, the zip file will be on the SM instance and not on the user's local filesystem. Seems like a step needs to be added above (where the .zip file is created) to download the zip locally.

Reply:
Added steps to outline the .ZIP interactions and removed wording for local use.